src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILImporter.Interpreter.cs

```diff
@@ -239,22 +269,914 @@ private void ImportLoadFloat(double value)
+                                break;
+                            case ILOpcode.clt:
+                                result = (long)val2 < (long)val1;
+                                break;
+                            case ILOpcode.clt_un:
+                                result = (long)((UIntPtr)val2.ToPointer()) < (long)((UIntPtr)val1.ToPointer());
+                                break;
+                            default:
+                                Debug.Assert(false);
+                                break;
+                        }
+                    }
+                    break;
+                case StackValueKind.Float:
+                    {
+                        if (op1.Kind < StackValueKind.Float || op2.Kind < StackValueKind.Float)
+                        {
+                            ThrowHelper.ThrowInvalidProgramException();
+                        }
+
+                        double val1 = op1.AsDouble();
+                        double val2 = op2.AsDouble();
+
+                        switch (opcode)
+                        {
+                            case ILOpcode.ceq:
+                                result = val1 == val2;
+                                break;
+                            case ILOpcode.cgt:
+                                result = val2 > val1;
+                                break;
+                            case ILOpcode.cgt_un:
+                                result = val2 > val1;
+                                break;
+                            case ILOpcode.clt:
+                                result = val2 < val1;
+                                break;
+                            case ILOpcode.clt_un:
+                                result = val2 < val1;
+                                break;
+                            default:
+                                Debug.Assert(false);
+                                break;
+                        }
+                    }
+                    break;
+                case StackValueKind.ObjRef:
+                    {
+                        if (op1.Kind < StackValueKind.ObjRef || op2.Kind < StackValueKind.ObjRef)
+                        {
+                            ThrowHelper.ThrowInvalidProgramException();
+                        }
+
+                        object val1 = op1.AsObjectRef();
+                        object val2 = op2.AsObjectRef();
+
+                        if (opcode == ILOpcode.ceq)
+                        {
+                            result = val1 == val2;
+                        }
+                        else
+                        {
+                            // TODO: Find GC addresses of objects and compare them
+                            throw new NotImplementedException();
+                        }
+                    }
+                    break;
+                case StackValueKind.Unknown:
+                case StackValueKind.ByRef:
```

Comment:
Is ByRef actually invalid here? I'd think you could at least compare it to null and maybe to another byref